### PR TITLE
Sunday efforts to get the motors fully working.

### PR DESCRIPTION
**Pull request by markstev**

### CODE REVIEW

File: arduino/motor_module.h [hunk 1]
```diff
@@ -101,10 +107,13 @@ class MotorModule : public arduinoio::UCModule {
       // steps (int32)
       // max_frequency (int32) = 1 / min_wait_
       char dir_pin = command[MOVE_LENGTH];
-      pulse_pin_ = command[MOVE_LENGTH + 1];
-      trigger_negative_pin_ = command[MOVE_LENGTH + 2];
-      trigger_positive_pin_ = command[MOVE_LENGTH + 3];
-      done_pin_ = command[MOVE_LENGTH + 4];
+      bool changed = Update(command[MOVE_LENGTH + 1], &pulse_pin_);
```
Comment (funrollloops) [inline arduino/motor_module.h:110]: if these members are initialized to something stable (-1?), initialized_ will be redundant.

inline initialization is clearer IMO, you could enable C++11 features by adding
target_compile_features( kamikaze PRIVATE cxx_constexpr cxx_decltype_auto )

to the CMakeLists.txt.

File: main.py [hunk 2]
```diff
@@ -185,19 +193,25 @@ def process_images():
           latest = q.get_nowait()
         except Queue.Empty:
           break
-      recognizer.detect_and_show(latest)
+      try:
+        recognizer.detect_and_show(latest)
+      except cv2.error:
+        # Sometimes, cvtcolor fails.
```
Comment (funrollloops) [inline main.py:199]: I think this only happens if the webcam/file failed to open -- should be unnecessary with the explicit check you added above.

File: main.py [hunk 3]
```diff
@@ -66,6 +66,7 @@ def __init__(self, robot):
     self.smile_cascade = cv2.CascadeClassifier(
         'haarcascades/haarcascade_smile.xml')
     self.robot = robot
+    self.last_action_time = time.time()
```
Comment (funrollloops) [inline main.py:69]: shouldn't we initialize on startup? i.e. set last_action_time to 0?

**Review by funrollloops — APPROVED** [review]
I'll make the changes myself, thanks for getting this to work!

What exactly did you have to change to get the movement to be smooth btw? The arduino code and the host both seem basically the same to me.

File: main.py (hunk 2 above)
Comment (markstev) [reply main.py:199]: I added this after adding the explicit check -- it happened after everything had been running for a couple of minutes. Seemed like a good thing to do to boost reliability.

File: arduino/motor_module.h (hunk 1 above)
Comment (markstev) [reply arduino/motor_module.h:110]: Good call.

File: main.py (hunk 3 above)
Comment (markstev) [reply main.py:69]: I already calibrate the motors when they're constructed. I suppose you could do that here instead, but that would make it possible to start detecting at a funny angle before calibrating.

**Pull request merged**